src/optimizer/physical_operators.cpp

```diff
@@ -413,6 +413,54 @@ bool InnerHashJoin::operator==(const BaseOperatorNodeContents &r) {
   return true;
 }
 
+
+
+//===--------------------------------------------------------------------===//
+// LeftSemiHashJoin
+//===--------------------------------------------------------------------===//
+BaseOperatorNodeContents *LeftSemiHashJoin::Copy() const { return new LeftSemiHashJoin(*this); }
+
+Operator LeftSemiHashJoin::Make(std::vector<AnnotatedExpression> &&join_predicates,
+                             std::vector<common::ManagedPointer<parser::AbstractExpression>> &&left_keys,
+std::vector<common::ManagedPointer<parser::AbstractExpression>> &&right_keys) {
+auto join = std::make_unique<LeftSemiHashJoin>();
+join->join_predicates_ = std::move(join_predicates);
```

Comment:
I think this will call the copu constructor. You should pass these in the constructor itself and initialize in the initializor list.